src/sources/community.js

```diff
@@ -0,0 +1,148 @@
+        console.log(`Error interpreting the default branch of ${this.constructor.name} for ${this.owner}/${this.repoName}`);
+        return "master";
+      });
+    if (!this.branch) {
+      this.branch = this.defaultBranch;
+    }
+  }
+
+  static get _name() { return "community"; }
+  get repo() { return `${this.owner}/${this.repoName}`; }
+  async baseUrl() {
+    return `https://github.com/${this.repo}/raw/${await this.branch}/`;
+  }
+
+  async repoNameWithBranch() {
+    const branch = await this.branch;
+    const defaultBranch = await this.defaultBranch;
+    if (branch === defaultBranch && !this.branchExplicitlyDefined) {
+      return this.repoName;
+    }
+    return `${this.repoName}@${branch}`;
+  }
+
+  dataset(pathParts) {
+    return new CommunityDataset(this, pathParts);
+  }
+  narrative(pathParts) {
+    return new CommunityNarrative(this, pathParts);
+  }
+
+  async availableDatasets() {
+    const qs = queryString.stringify({ref: await this.branch});
+    const response = await fetch(`https://api.github.com/repos/${this.repo}/contents/auspice?${qs}`);
+
+    if (response.status === 404) throw new NotFound();
+    else if (response.status !== 200 && response.status !== 304) {
+      utils.warn(`Error fetching available datasets from GitHub for source ${this.name}`, await utils.responseDetails(response));
+      return [];
+    }
+
+    const filenames = (await response.json())
+      .filter((file) => file.type === "file")
+      // remove anything which doesn't start with the repo name, which is required of community datasets
+      .filter((file) => file.name.startsWith(this.repoName))
+      .map((file) => file.name);
+    const pathnames = utils.getDatasetsFromListOfFilenames(filenames)
+      // strip out the repo name from the start of the pathnames
+      // as CommunityDataset().baseParts will add this in
+      .map((pathname) => pathname.replace(`${this.repoName}/`, ""));
+    return pathnames;
+  }
+
+  async availableNarratives() {
+    const qs = queryString.stringify({ref: await this.branch});
+    const response = await fetch(`https://api.github.com/repos/${this.repo}/contents/narratives?${qs}`);
+
+    if (response.status !== 200 && response.status !== 304) {
+      if (response.status !== 404) {
+        // not found doesn't warrant an error print, it means there are no narratives for this repo
+        utils.warn(`Error fetching available narratives from GitHub for source ${this.name}`, await utils.responseDetails(response));
+      }
+      return [];
+    }
+
+    const files = await response.json();
+    return files
+      .filter((file) => file.type === "file")
+      .filter((file) => file.name !== "README.md")
+      .filter((file) => file.name.endsWith(".md"))
+      .filter((file) => file.name.startsWith(this.repoName))
+      .map((file) => file.name
+        .replace(this.repoName, "")
+        .replace(/^_/, "")
+        .replace(/[.]md$/, "")
+        .split("_")
+        .join("/"));
```

Comment:
This logic seems pretty important from a user perspective. Is it documented anywhere?

(I know this is outside the PR scope of splitting src/sources.js into smaller files – thanks for doing that!)

Reply:
Community filename details documented at https://docs.nextstrain.org/en/latest/guides/share/community-builds.html#technical-details